spring-batch-docs/asciidoc/spring-batch-integration.adoc

```diff
@@ -1104,125 +1053,100 @@ configuration:
-	return stepExecutionRequestHandler();
+public StepExecutionRequestHandler serviceActivator() throws Exception {
+    return stepExecutionRequestHandler();
 }
 
 @Bean
 public DirectChannel outboundStaging() {
-	return new DirectChannel();
+    return new DirectChannel();
 }
 
 @Bean
-@ServiceActivator(inputChannel = "outboundStaging")
-public MessageHandler outboundStagingJmsSendingHandler() {
-	JmsSendingMessageHandler handler = new JmsSendingMessageHandler(jmsTemplate());
-	handler.setDestinationName("stagingQueue");
-	return handler;
+public IntegrationFlow outboundJmsStaging() {
+    return IntegrationFlows.from("outboundStaging")
+            .handle(Jms.outboundGateway(connectionFactory())
+                    .requestDestination("stagingQueue"))
+            .get();
 }
 
 @Bean
 public DirectChannel inboundStaging() {
-	return new DirectChannel();
-}
-
-@Bean
-public DefaultMessageListenerContainer inboundStagingListenerContainer() {
-	DefaultMessageListenerContainer defaultMessageListenerContainer = new DefaultMessageListenerContainer();
-	defaultMessageListenerContainer.setDestinationName("stagingQueue");
-	return defaultMessageListenerContainer;
-}
-
-@Bean
-public ChannelPublishingJmsMessageListener inboundStagingJmsMessageListener() {
-	ChannelPublishingJmsMessageListener channelPublishingJmsMessageListener = new  ChannelPublishingJmsMessageListener();
-	return channelPublishingJmsMessageListener;
+    return new DirectChannel();
 }
 
 @Bean
-public JmsMessageDrivenEndpoint inboundStagingAdapter() {
-	JmsMessageDrivenEndpoint jmsMessageDrivenEndpoint = new JmsMessageDrivenEndpoint(inboundStagingListenerContainer(), inboundStagingJmsMessageListener());
-	jmsMessageDrivenEndpoint.setOutputChannel(inboundStaging());
-	return jmsMessageDrivenEndpoint;
+public IntegrationFlow inboundJmsStaging() {
+    return IntegrationFlows.from(Jms.inboundAdapter(connectionFactory())
+                    .configureJmsTemplate(t -> t.deliveryPersistent(true)
+                            .jmsMessageConverter(new SimpleMessageConverter()))
+                    .destination("stagingQueue"), c -> c.poller(Pollers.fixedRate(1000).maxMessagesPerPoll(1)))
+            .channel(inboundStaging())
+            .get();
 }
 
 @Bean
 @ServiceActivator(inputChannel = "inboundStaging")
-public MessageHandler partitioningMessageHandler() throws Exception{
-	AggregatorFactoryBean aggregatorFactoryBean = new AggregatorFactoryBean();
-	aggregatorFactoryBean.setProcessorBean(partitioningMessageHandler());
-	aggregatorFactoryBean.setOutputChannel(outboundReplies());
-	...
-	return aggregatorFactoryBean.getObject();
+public MessageHandler partitioningMessageHandler() throws Exception {
+    AggregatorFactoryBean aggregatorFactoryBean = new AggregatorFactoryBean();
+    aggregatorFactoryBean.setProcessorBean(partitionHandler());
+    aggregatorFactoryBean.setOutputChannel(outboundReplies());
+    ...
+    return aggregatorFactoryBean.getObject();
```

Comment:
No, you don't need to do this manually. The Framework understands properly that `FactoryBean` is goo there.

Or has that stopped to work recently since you've been forced to do this?

Reply:
Updated.  It works perfectly, I just made a mistake.